### DISCUSSION TITLE
Injection and extraction cycles

### DISCUSSION THREAD

**Discussion by clariu25** [idaholab/moose#27218] [Q&A Modules: Porous Flow]
Hello!
I'm trying to define a geothermal reservoir using Porous flow. In the current file I have only 1 well that is producing. For that I'm using the PorousFlowPolyLineSink, for both mass and heat extraction. I allow the system to reach equilibrium and the Dirac Kernels only start after t=0. I'm using a ConditionalFunctionEnableControl for this

For the production, I set the flow positive, to extract. I see a decrease of the pressure around the well but, for some reason, the temperature increases around the well for the first steps. Later, the temperature decreases around the well, increases a bit further and then comes back to the initial temperature. Shouldn't I only see decrease of temperature?

```c++
[DiracKernels]
  [production_T]
    type = PorousFlowPolyLineSink
    variable = temperature
    SumQuantityUO = extracted_heat
    point_file = point_130m.bh
    fluid_phase = 0
    p_or_t_vals = 0
    fluxes = 0.6 # 2Kg/s over 6m well
    use_enthalpy = true
  []
  [production_P]
    type = PorousFlowPolyLineSink
    variable = porepressure
    SumQuantityUO = extracted_mass
    point_file = point_130m.bh
    p_or_t_vals = 0
    fluxes = 0.3 # 2Kg/s over 6m well
  []
[]
``` 

![image](https://github.com/idaholab/moose/assets/75475389/fac98bb2-46ab-4b11-afcc-1c9ed5bf4995)

I'm attaching the input and the point files.

I would really appreciate if someone could take a look at this. Thank you
[PF_THM_Prod_1yr.zip](https://github.com/idaholab/moose/files/14794507/PF_THM_Prod_1yr.zip)

**Comment by GiudGiud** [comment]
Hello

This doesnt look too physical indeed.
Is the result converged?
I see 
```
  nl_rel_tol = 1e-6
  nl_rel_step_tol = 1e-4
```
in the executioner which is potentially accepting unconverged solutions. Especially with automatic scaling as the residual for each variables are scaled 

Is this something you could solve without the mechanics part first? To see if the temperature problem is well converged

Guillaume

**Reply by clariu25** [reply]
Hello! Thanks for replying and for the suggestions.

The model seemed to be converging. Most NL values were in the 1e-5 to 1e-8 orders, and the temperature residuals were much lower than the pressure ones.

I removed the mechanics part and now I'm only doing TH and I also reduced the tolerance. Still, I'm getting the same result.
```c++
  nl_rel_tol = 1e-8
  nl_rel_step_tol = 1e-8
  l_tol = 1e-8
``` 
![image](https://github.com/idaholab/moose/assets/75475389/34ec6da0-22a0-4714-80da-685f632b0540)

I reduced the element size, just to test it and it did not have much effect. By removing the automatic scaling, the convergence is better, both variable residuals are in the order of 1e-12 now but the issue with the temperature is still there. 

For what I read in some other discussions, the `PorousFlowPolyLineSink` is a very good option to simulate geothermal production, right? Or is there a better solution for the heat extraction? I know that for injection one should use other options.

Thank you again

[PF_TH_Prod.zip](https://github.com/idaholab/moose/files/14872497/PF_TH_Prod.zip)

**Comment by Rahim-Habibi** [comment]
Hi @clariu25 

First, i would recommend to run the model with only one injection point, to make it easy for understanding and interpretation. 
Have you tried to apply heat source using BC rather than direckernel? 
Rahim
